src/app.js

```diff
@@ -1,80 +1,86 @@
-require('dotenv').config();
-
-const express = require('express');
-const session = require('express-session');
-const app = express();
-
-const morgan = require('morgan');
-const bodyParser = require('body-parser');
-const cors = require('cors');
-const favicon = require('express-favicon');
-const xss = require('xss-clean');
-const helmet = require('helmet');
-
-const passport = require('passport');
-const mongoose = require('mongoose');
-mongoose.set('strictQuery', true);
-const asyncErrors = require('express-async-errors');
-const cookieParser = require('cookie-parser');
-const rateLimit = require('express-rate-limit');
+require('dotenv').config()
+
+const express = require('express')
+const session = require('express-session')
+const app = express()
+
+const morgan = require('morgan')
+const bodyParser = require('body-parser')
+const cors = require('cors')
+const favicon = require('express-favicon')
+const xss = require('xss-clean')
+const helmet = require('helmet')
+
+const passport = require('passport')
+const mongoose = require('mongoose')
+mongoose.set('strictQuery', true)
+const asyncErrors = require('express-async-errors')
+const cookieParser = require('cookie-parser')
+const rateLimit = require('express-rate-limit')
 const limiter = rateLimit({
   windowMs: 15 * 60 * 1000, // 15 minutes
-  max: 60, // limit each IP to 100 requests per windowMs
-});
+  max: 60 // limit each IP to 100 requests per windowMs
+})
+const logger = require('./logger')
```

Comment:
Where is this file or folder? I don't see it checked into this PR

Reply:
For a simple Winston logger that prints to the console/stdout, you could have a file like `src/logger/index.js`:

```javascript
const { createLogger, format, transports } = require('winston');

const logger = createLogger({
  level: 'info',
  format: format.combine(
    // Sets up the padding and coloring of the logs for the console
    format.cli(),
    // Allows us to log errors directly, eg: logger.error(new Error('Error Message')) will log the stack trace for that error
    format.errors({ stack: true }),
    // generates a timestamp that we can add to each log
    format.timestamp({
      format: 'YYYY-MM-DD HH:mm:ss'
    }),
    format.printf(({ level, message, timestamp, stack }) => { 
      // Formatting the output of the logging function.
      // Stack may be undefined if there was no error passed to this log.
      return stack ? `[${timestamp} ${level}]: ${message} - ${stack}` : `[${timestamp}] ${level}: ${message}`;
    })
  ),
  transports: [
    new transports.Console({'timestamp':true}),
    // If we wanted to write logs to a file, we could uncomment below:
    // new transports.File({ filename: 'quick-start-combined.log' })
  ]
});

module.exports = logger;
```

Reply:
for winston I have created folder logs with file logger.js 
`// logger.js
const winston = require("winston");

// Define the log levels and colors (optional)
const logLevels = {
  error: "red",
  warn: "yellow",
  info: "green",
  debug: "blue",
};

winston.addColors(logLevels);

const logger = winston.createLogger({
  level: "debug",
  levels: logLevels,
  format: winston.format.combine(
    winston.format.colorize(),
    winston.format.timestamp({ format: "YYYY-MM-DD HH:mm:ss" }),
    winston.format.printf(({ timestamp, level, message }) => {
      return `${timestamp} [${level.toUpperCase()}]: ${message}`;
    }),
  ),
  transports: [
    new winston.transports.Console(),
    new winston.transports.File({ filename: "logs/app.log" }),
  ],
});

module.exports = logger;
`

Reply:
Should I change for your code ?

Reply:
Ah what you have seems fine as well; I am not seeing it in this PR however?

Reply:
It is strange, I have this folder in root directory in my branch. Should I try to change the location of the folder logs possibly ?

Reply:
You try doing `git add ./logger` and see if it shows up in "Changes staged for commit" when you do `git status` ? 

I'm wondering if this `.gitignore` file is causing it to be ignored, though that would surprise me since it doesn't match up to what you said you've called it.

Reply:
Akos you are completely right , this folder logs with logger.js  was ignored by .gitignore. Now I pushed it to my branch. Could you please check if you see it now ? commit - cdbceb8eb06c70fb5ecd5f5e18cc2d25cf496bf8

Reply:
Yes, I can see the file now :tada: 

This line should be updated though to match the path the file is at otherwise Node won't be able to find/load it correctly:

```suggestion
const logger = require('./logs/logger')
```